### PR TITLE
[api] Remove 8.10 deprecations.

### PR DESCRIPTION
**Pull request by ejgallego**
Almost ready, pending PRs:

- https://github.com/HoTT/HoTT/pull/1029
- https://github.com/Mtac2/Mtac2/pull/183

### CODE REVIEW

File: proofs/proof.ml [hunk 1]
```diff
@@ -554,3 +508,19 @@ let data { proofview; focus_stack; entry; shelf; given_up; initial_euctx; name;
...
+      | [] -> acc
+      | (before,after)::stack ->
+         aux (pr_goal_list before ++ spc () ++ str "{" ++ acc ++ str "}" ++ spc () ++
+              pr_goal_list after) stack in
+    str "[" ++ str "focus structure: " ++
+               aux (pr_goal_list fg_goals) bg_goals ++ str ";" ++ spc () ++
+    str "shelved: " ++ pr_goal_list shelf ++ str ";" ++ spc () ++
+    str "given up: " ++ pr_goal_list given_up ++
+    str "]"
+  )
```
Comment (SkySkimmer) [inline proofs/proof.ml]: Why's this function moving in this file?

Comment (ejgallego) [reply proofs/proof.ml]: Because it depends on a function introduced later on.

**Review by SkySkimmer — APPROVED** [review]
Needs some overlays.

**Comment by ejgallego** [comment]
> Needs some overlays.

Thanks, will wait a few days so hopefully the 8.10 branch is more stable.

**Comment by ejgallego** [comment]
I had to restore `nf_enter` as mtac2 is broken otherwise; no major problems it seems except for a trivial tweak needed for hott (and merging of https://github.com/Mtac2/Mtac2/pull/183)

**Comment by SkySkimmer** [comment]
Still waiting on the mtac2 pr.

**Comment by ejgallego** [comment]
Should be fixed.

**Comment by ejgallego** [comment]
Should be good now [I hope]

**Comment by ejgallego** [comment]
Indeed, ready to go.

**Pull request merged**